
Status: Rejected in upstream (kern/77353)

postgresql documentation encourages to grow SysV IPC limits on *BSD systems
(which obviously applies to GNU/k*BSD ones too):

<quote>
    The options SYSVSHM and SYSVSEM need to be enabled when the kernel is compiled. (They are by default.) The maximum size of shared memory is determined by the option SHMMAXPGS (in pages). The following shows an example of how to set the various parameters:

options         SYSVSHM
options         SHMMAXPGS=4096
options         SHMSEG=256

options         SYSVSEM
options         SEMMNI=256
options         SEMMNS=512
options         SEMMNU=256
options         SEMMAP=256

    (On NetBSD and OpenBSD the key word is actually option singular.)

    You might also want to configure your kernel to lock shared memory into RAM and prevent it from being paged out to swap. Use the sysctl setting kern.ipc.shm_use_phys. 
</quote>

Index: sys/i386/conf/GENERIC
===================================================================
--- sys/i386/conf/GENERIC.orig	2007-12-09 13:24:47.000000000 +0100
+++ sys/i386/conf/GENERIC	2007-12-09 13:25:00.391532000 +0100
@@ -55,8 +55,14 @@
 options 	SCSI_DELAY=5000		# Delay (in ms) before probing SCSI
 options 	KTRACE			# ktrace(1) support
 options 	SYSVSHM			# SYSV-style shared memory
+options         SHMMAXPGS=4096
+options         SHMSEG=256
 options 	SYSVMSG			# SYSV-style message queues
 options 	SYSVSEM			# SYSV-style semaphores
+options         SEMMNI=256
+options         SEMMNS=512
+options         SEMMNU=256
+options         SEMMAP=256
 options 	_KPOSIX_PRIORITY_SCHEDULING # POSIX P1003_1B real-time extensions
 options 	KBD_INSTALL_CDEV	# install a CDEV entry in /dev
 options 	ADAPTIVE_GIANT		# Giant mutex is adaptive.
Index: sys/amd64/conf/GENERIC
===================================================================
--- sys/amd64/conf/GENERIC.orig	2007-12-09 13:24:47.000000000 +0100
+++ sys/amd64/conf/GENERIC	2007-12-09 13:25:00.320142000 +0100
@@ -55,8 +55,14 @@
 options 	SCSI_DELAY=5000		# Delay (in ms) before probing SCSI
 options 	KTRACE			# ktrace(1) support
 options 	SYSVSHM			# SYSV-style shared memory
+options         SHMMAXPGS=4096
+options         SHMSEG=256
 options 	SYSVMSG			# SYSV-style message queues
 options 	SYSVSEM			# SYSV-style semaphores
+options         SEMMNI=256
+options         SEMMNS=512
+options         SEMMNU=256
+options         SEMMAP=256
 options 	_KPOSIX_PRIORITY_SCHEDULING # POSIX P1003_1B real-time extensions
 options 	KBD_INSTALL_CDEV	# install a CDEV entry in /dev
 options 	ADAPTIVE_GIANT		# Giant mutex is adaptive.
